client/app/pages/dashboards/ShareDashboardDialog.jsx

```diff
@@ -0,0 +1,122 @@
+import { replace } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import Switch from 'antd/lib/switch';
+import Modal from 'antd/lib/modal';
+import Form from 'antd/lib/form';
+import Tooltip from 'antd/lib/tooltip';
+import { $http, toastr } from '@/services/ng';
+import { wrap as wrapDialog, DialogPropType } from '@/components/DialogWrapper';
+import InputWithCopy from '@/components/InputWithCopy';
+
+const API_SHARE_URL = 'api/dashboards/{id}/share';
+const HELP_URL = 'https://redash.io/help/user-guide/dashboards/sharing-dashboards?source=dialog';
+
+class ShareDashboardDialog extends React.Component {
+  static propTypes = {
+    dashboard: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    dialog: DialogPropType.isRequired,
+  };
+
+  formItemProps = {
+    labelCol: { span: 8 },
+    wrapperCol: { span: 16 },
+    style: { marginBottom: 7 },
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      saving: false,
+      apiUrl: replace(API_SHARE_URL, '{id}', props.dashboard.id),
+    };
+  }
+
+  static get headerContent() {
+    return (
+      <React.Fragment>
+        Share Dashboard
+        <div className="modal-header-desc">
+          Allow public access to this dashboard with a secret address.{' '}
+          <Tooltip title="Guide: Sharing and Embedding Dashboards">
+            { /* eslint-disable-next-line react/jsx-no-target-blank */}
+            <a href={HELP_URL} target="_blank" rel="noopener">Learn more</a>
+          </Tooltip>
+        </div>
+      </React.Fragment>
+    );
+  }
+
+  enable = () => {
+    const { dashboard } = this.props;
+    this.setState({ saving: true });
+
+    $http
```

Comment:
This is more like a question, "how" is it ok to use `$http` inside components?

Reply:
Not really ok. @ranbena you can add some methods to `services/dashboard`:
```javascript
const resource = $resource(
  // ...
  share: {
    method: 'post',
    url: 'api/dashboards/:id/share',
  },
  unshare: {
    method: 'delete',
    url: 'api/dashboards/:id/share',
  },
);
```
Usage:
```javascript
this.props.dashboard.share()
  .then(...)
  .catch(...)
  .finally(...);
```

Reply:
While it make sense to have a layer where we implement the interaction with the server and expose a simple API to the rest of application, I don't think we should invest in implementing any more code with $resource...

Let's leave as is, and refactor when we decide on the $resource/$http successor.